watchtower/wtwire/state_update.go

```diff
@@ -0,0 +1,81 @@
+package wtwire
+
+import "io"
+
+// StateUpdate transmits an encrypted state update from the client to the
+// watchtower. Each state update is tied to particular session, identified by
+// the client's brontine key used to make the request.
+type StateUpdate struct {
+	// SeqNum is a 1-indexed, monotonically incrementing sequence number.
+	// This number should always be one equal to the total number of
+	// successful updates sent to the watchtower. This value must always be
+	// less or equal than the negotiated MaxUpdates for the session.
+	SeqNum uint16
```

Comment:
Why 1 indexed?

Reply:
The smallest `LastApplied` we can send is 0, and signals that we've already received an ACK from the tower for that sequence number. Thus, we cannot send sequence number 0 for the first time and already have an ACK for 0. Another interpretation would be to think of sequence number 0 as the `SessionInit` message.

Reply:
👍